week-1-ts-ecosystem/w01e02-enter-the-world-of-typescript/src/framework/dom-creators.ts

```diff
@@ -0,0 +1,16 @@
+type TagName = 'p' | 'div' | 'section' | 'article';
+
+function createElement(tagName: TagName = 'div', className: string = ''): HTMLElement {
```

Comment:
Niektóre z elementów `TagName` powinny bardziej precyzyjnie zwracać Typ klasy.
np: `HTMLElement` | `HTMLDivElement` oraz `HTMLParagraphElement`

Osiągniesz to dzięki dorzuceniu typu generycznego rozszerzającego `TagName` do tej funkcji.